prometheus/utils_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package prometheus_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// removeBrittleJSON is a helper function to compare proto messages in json format.
+// Without removing brittle, we can't assert that two proto messages in json/text format are equal.
+// Read more in https://github.com/golang/protobuf/issues/1121
+func removeBrittleJSON(j []byte) string {
```

Comment:
Not sure where you got this "brittle" word from. (: Let's be specifc and call it "normalization" rather?

Also printing errors is odd if nothing guards callers from using this function for other things than printing.

Finally, we have opportunity to reduce boilerplate of marshalling too and make this function generally type safe. We know we want to use it only for printing in example, let's block other use cases, so further devs will be forced to produce golden tests instead (: 

I would propose:

```suggestion
func printlnNormalized(t testing.TB, msg proto.Message) {
  t.Helper()
  
   // marshall, do t.Fatal on error
   
   // compact json
   
   // print to stdout (e.g. fmt.Println)
}
```

WDYT?

Reply:
> Not sure where you got this "brittle" word from. (: Let's be specifc and call it "normalization" rather?

I guess I'm not 100% fluent in English and sometimes some words just get stuck in my head 😅. Agreed that normalization sounds better
 
> Also printing errors is odd if nothing guards callers from using this function for other things than printing.
>
> Finally, we have opportunity to reduce boilerplate of marshalling too and make this function generally type safe. We know we want to use it only for printing in example, let's block other use cases, so further devs will be forced to produce golden tests instead (:
> 
> I would propose:
> 
> WDYT?

Sounds good! Thanks for the suggestions

Reply:
Actually, examples don't have parameter testing.T, so calling 'printlnNormalized(**t testing.TB**...' doesn't seem possible?

I'm panic'ing instead of calling t.Fatal, but let me know if I missed something here

Reply:
👍🏽 Good point, panic is fine